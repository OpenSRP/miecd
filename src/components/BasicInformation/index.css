#detailsRow {
    width: 100%;
}

#detailsCard {
    width: 100%;
}
/* 
#detailsCard .row {
    margin: 0 !important;
} */

.card-title {
    background-color: #ffffff;
    color: #3a484f;
    border-bottom: 1px solid #dadce0;
    font-family: Roboto;
    font-size: 1.7rem;
    font-weight: 500;
    letter-spacing: -0.018rem;
    line-height: 2rem;
    text-align: left;
    padding: 0.5em 1rem;
}

.information__card_body_cell {
    width: 25%;
    min-width: 250px;
}

.information__card_body_cell .prop {
    color: #637381;
    margin-right: 20px;
}

.prop,
.value {
    font-size: 1rem;
}
