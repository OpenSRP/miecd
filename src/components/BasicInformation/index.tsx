import React from 'react';
import { useTranslation } from 'react-i18next';
import { Card, CardTitle, Col, Row, Table } from 'reactstrap';
import './index.css';

export interface LabelValuePair {
    label: string;
    value: string | number;
}

interface Props {
    labelValuePairs: LabelValuePair[];
}

export default function BasicInformation({ labelValuePairs = [] }: Props) {
    const { t } = useTranslation();
    return (
        <Row id="detailsRow">
            <Card id="detailsCard">
                <CardTitle>{t('Basic Information')}</CardTitle>
                <Row>
                    <Col className="detailsColumn" sm="12" lg="6">
                        <Table borderless>
                            <tbody>
                                {labelValuePairs.map((labelValuePair: LabelValuePair, index: number) => {
                                    if (index % 2) {
                                        return (
                                            <tr key={index}>
                                                <td>{labelValuePair.label}</td>
                                                <td>{labelValuePair.value}</td>
                                            </tr>
                                        );
                                    }
                                    return null;
                                })}
                            </tbody>
                        </Table>
                    </Col>
                    <Col className="detailsColumn" sm="12" lg="6">
                        <Table borderless>
                            <tbody>
                                {labelValuePairs.map((labelValuePair: LabelValuePair, index: number) => {
                                    if (!(index % 2)) {
                                        return (
                                            <tr key={index}>
                                                <td>{labelValuePair.label}</td>
                                                <td>{labelValuePair.value}</td>
                                            </tr>
                                        );
                                    }
                                    return null;
                                })}
                            </tbody>
                        </Table>
                    </Col>
                </Row>
            </Card>
        </Row>
    );
}
