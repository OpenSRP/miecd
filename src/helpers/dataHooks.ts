import {
    OPENSRP_SECURITY_AUTHENTICATE,
    LOCATION_FILTER_PARAM,
    RISK_CATEGORY_FILTER_PARAM,
    SMS_TYPE_FILTER_PARAM,
    SEARCH_FILTER_PARAM,
    PREGNANCY_DETECTION,
    UPDATE_PREGNANCY_DETECTION,
    NUTRITION_REGISTRATION,
    UPDATE_NUTRITION_REGISTRATION,
    DEATH_REPORT,
    UPDATE_DEATH_REPORT,
} from '../constants';
import { getQueryParams } from 'containers/LogFace/utils';
import { TFunction } from 'i18next';
import { useQuery } from 'react-query';
import { useSelector } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { OpenSRPService } from 'services/opensrp';
import { getSmsDataByFilters } from 'store/ducks/sms_events';
import { TreeNode } from './locationHierarchy/types';
import { generateJurisdictionTree } from './locationHierarchy/utils';
import { toastToError, getRiskCatFilter } from './utils';
import { LogFaceModules } from 'configs/settings';

const selectSmsData = getSmsDataByFilters();

/** hook to get this users assignment details, hierarchy, team assigned and location the team is assigned
 * @param t - the translator function
 */
export const useUserAssignment = (t: TFunction) => {
    return useQuery(
        'userAssignment',
        () => {
            const opensrpService = new OpenSRPService(OPENSRP_SECURITY_AUTHENTICATE);
            return opensrpService.read('');
        },
        {
            select: (response) => {
                return {
                    userHierarchy: generateJurisdictionTree(response.locations),
                    user: response.user,
                    team: response.team.team,
                    location: response.team.team.location,
                };
            },
            onError: (err: Error) => toastToError(err?.message ?? t('An error occurred')),
        },
    );
};

/** redux hok to select filtered sms data based on provided filters
 * @param location - route location details, to get search params
 * @param module - the module
 * @param userLocationTree - for filtering based on the user's assigned level
 */
export const useGetLogFaceFilteredSms = (
    location: RouteComponentProps['location'],
    module: LogFaceModules,
    userHierarchy?: TreeNode,
) => {
    return useSelector((state) => {
        const userLocationIdFilter = getQueryParams(location)[LOCATION_FILTER_PARAM] as string;
        const riskCategoryFilter = getQueryParams(location)[RISK_CATEGORY_FILTER_PARAM] as string;
        const smsTypeFilter = getQueryParams(location)[SMS_TYPE_FILTER_PARAM] as string;
        const searchFilter = getQueryParams(location)[SEARCH_FILTER_PARAM] as string;
        const matchesId = (node: TreeNode) => node.model.id === userLocationIdFilter;
        const userLocationNode = userHierarchy?.first(matchesId);
        let smsFilter = smsTypeFilter ? [smsTypeFilter] : undefined;
        switch (smsTypeFilter) {
            case PREGNANCY_DETECTION:
                smsFilter = [PREGNANCY_DETECTION, UPDATE_PREGNANCY_DETECTION];
                break;
            case NUTRITION_REGISTRATION:
                smsFilter = [NUTRITION_REGISTRATION, UPDATE_NUTRITION_REGISTRATION];
                break;
            case DEATH_REPORT:
                smsFilter = [DEATH_REPORT, UPDATE_DEATH_REPORT];
                break;
            default:
                break;
        }
        return selectSmsData(state, {
            locationNode: userLocationNode,
            riskCategory: getRiskCatFilter(module, riskCategoryFilter),
            smsTypes: smsFilter,
            searchFilter,
            module,
        });
    });
};
