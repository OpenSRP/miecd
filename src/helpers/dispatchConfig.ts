import { ConfigState, setAllConfigs, getAllConfigs } from '@opensrp/pkg-config';
import { OPENSRP_REST_API_BASE_URL } from 'configs/settings';
import { BACKEND_ACTIVE } from '../configs/env';
import { BACKEND_LOGIN_URL, REACT_LOGIN_URL } from '../constants';
import i18n from '../mls';

export const APP_LOGIN_URL = BACKEND_ACTIVE ? BACKEND_LOGIN_URL : REACT_LOGIN_URL;

const defaultvalues = getAllConfigs();

const configObject: ConfigState = {
    ...defaultvalues,
    appLoginURL: APP_LOGIN_URL,
    opensrpBaseURL: OPENSRP_REST_API_BASE_URL,
    i18n,
};

setAllConfigs(configObject);
