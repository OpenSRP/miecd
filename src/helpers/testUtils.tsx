import { mount, MountRendererProps } from 'enzyme';
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import i18n from '../mls';
import { act } from 'react-dom/test-utils';

export const mountWithTranslations: typeof mount = (component: React.ReactNode, options?: MountRendererProps) => {
    return mount(<I18nextProvider i18n={i18n}>{component}</I18nextProvider>, options);
};

/**
 * Utility to await promises to resolve in tests
 * @param ms time in ms
 */
export async function waitForPromises() {
    await act(async () => {
        await new Promise((res) => setImmediate(res));
    });
}
