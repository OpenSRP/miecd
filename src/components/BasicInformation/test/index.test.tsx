import toJson from 'enzyme-to-json';
import BasicInformation from '..';
import { mountWithTranslations } from '../../../helpers/testUtils';

const props = [
    { label: 'currentEdd', value: 'test edd' },
    { label: 'currentGravidity', value: 20 },
    { label: 'currentParity', value: 20 },
    { label: 'id', value: 'test' },
    { label: 'location', value: 'Test Location' },
    { label: 'previousPregnancyRisk', value: 'no risk' },
];

describe('BasicInformation', () => {
    it('must render correctly', () => {
        const wrapper = mountWithTranslations(<BasicInformation labelValuePairs={props} />);
        expect(toJson(wrapper.find('BasicInformation'))).toMatchSnapshot();
    });
});
