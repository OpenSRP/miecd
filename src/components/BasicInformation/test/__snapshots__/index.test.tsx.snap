// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicInformation must render correctly 1`] = `
<BasicInformation
  labelValuePairs={
    Array [
      Object {
        "label": "currentEdd",
        "value": "test edd",
      },
      Object {
        "label": "currentGravidity",
        "value": 20,
      },
      Object {
        "label": "currentParity",
        "value": 20,
      },
      Object {
        "label": "id",
        "value": "test",
      },
      Object {
        "label": "location",
        "value": "Test Location",
      },
      Object {
        "label": "previousPregnancyRisk",
        "value": "no risk",
      },
    ]
  }
>
  <div
    id="detailsRow"
  >
    <Card
      id="detailsCard"
      tag="div"
    >
      <div
        className="card"
        id="detailsCard"
      >
        <CardTitle
          tag="div"
        >
          <div
            className="card-title"
          >
            Basic Information
          </div>
        </CardTitle>
        <div
          className="d-flex justify-content-between p-3 flex-wrap information__card_body"
        >
          <div
            className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
            key="currentEdd-test edd"
          >
            <span
              className="prop"
            >
               
              currentEdd
              :
            </span>
            <span
              className="value"
            >
              test edd
            </span>
          </div>
          <div
            className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
            key="currentGravidity-20"
          >
            <span
              className="prop"
            >
               
              currentGravidity
              :
            </span>
            <span
              className="value"
            >
              20
            </span>
          </div>
          <div
            className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
            key="currentParity-20"
          >
            <span
              className="prop"
            >
               
              currentParity
              :
            </span>
            <span
              className="value"
            >
              20
            </span>
          </div>
          <div
            className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
            key="id-test"
          >
            <span
              className="prop"
            >
               
              id
              :
            </span>
            <span
              className="value"
            >
              test
            </span>
          </div>
          <div
            className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
            key="location-Test Location"
          >
            <span
              className="prop"
            >
               
              location
              :
            </span>
            <span
              className="value"
            >
              Test Location
            </span>
          </div>
          <div
            className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
            key="previousPregnancyRisk-no risk"
          >
            <span
              className="prop"
            >
               
              previousPregnancyRisk
              :
            </span>
            <span
              className="value"
            >
              no risk
            </span>
          </div>
        </div>
      </div>
    </Card>
  </div>
</BasicInformation>
`;
